compute/auth/access_token.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of authenticating using access tokens directly on Compute Engine.
+
+For more information, see the README.md under /compute.
+"""
+
+# [START all]
+
+import argparse
+
+import requests
+
+
+METADATA_URL = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+SERVICE_ACCOUNT = 'default'
+
+
+def get_access_token():
+    url = '{}instance/service-accounts/{}/token'.format(
+        METADATA_URL, SERVICE_ACCOUNT)
+
+    # Request an access token from the metadata server.
+    r = requests.get(url, headers=METADATA_HEADERS)
+    r.raise_for_status()
+
+    # Extract the access token from the respose.
+    access_token = r.json()['access_token']
+
+    return access_token
+
+
+def list_buckets(project_id, access_token):
+    url = 'https://www.googleapis.com/storage/v1/b'
+    params = {
+        'project': project_id
+    }
+    headers = {
+        'Authenication': 'Bearer {}'.format(access_token)
```

Comment:
Phun caught it, and it's already fixed. :)

On Fri, Apr 22, 2016 at 1:52 PM elibixby <notifications@github.com> wrote:

> In compute/auth/access_token.py
> <https://github.com/GoogleCloudPlatform/python-docs-samples/pull/275#discussion_r60801450>
> :
>
> > +    r = requests.get(url, headers=METADATA_HEADERS)
> > +    r.raise_for_status()
> > +
> > +    # Extract the access token from the respose.
> > +    access_token = r.json()['access_token']
> > +
> > +    return access_token
> > +
> > +
> > +def list_buckets(project_id, access_token):
> > +    url = 'https://www.googleapis.com/storage/v1/b'
> > +    params = {
> > +        'project': project_id
> > +    }
> > +    headers = {
> > +        'Authenication': 'Bearer {}'.format(access_token)
>
> Typo!
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/python-docs-samples/pull/275/files/133b62278a12ed4c93dce46db710eb5598a919eb#r60801450>
>